### PR TITLE
fileio backstore: fix sparse file creation

### PR DESCRIPTION
**Pull request by maurizio-lombardi**
fallocate() can't be used to create sparse files because it
actually preallocates all the disk space that will be used by the file
backstore, sparse files do not have preallocated disk space
by definition.
We must therefore use ftruncate().

We can, on the other hand, use fallocate() to create non-sparse
files and fall back to the slower "while() fwrite()" if we
are running on Python version < 3.3 where fallocate() is not available

Fixes 3bd4d8ef7c9b154c53e8b8dd863a570bce7f5c2c

Signed-off-by: Maurizio Lombardi <mlombard@redhat.com>

### CODE REVIEW

File: targetcli/ui_backstore.py [hunk 1]
```diff
@@ -423,17 +423,17 @@ def _create_file(self, filename, size, sparse=True):
...
                     # Prior to version 3.3, Python does not provide fallocate
-                    os.ftruncate(f.fileno(), size)
-            else:
-                self.shell.log.info("Writing %d bytes" % size)
-                while size > 0:
-                    write_size = min(size, 1024)
-                    f.write("\0" * write_size)
-                    size -= write_size
+                    while size > 0:
+                        write_size = min(size, 1024)
```
Comment (pkalever) [inline targetcli/ui_backstore.py:434]: Do we only accept size in 1024 chucks only?
Say what if the user asks to create a storage object with a size aligned to 512? say request to create for 2560B?

Comment (maurizio-lombardi) [reply targetcli/ui_backstore.py:434]: @pkalever No, it calls min() so with your example the while() will do 3 cycles:
two cycles with write_size = 1024 and the last one with write_size = min(512, 1024) = 512

Comment (pkalever) [reply targetcli/ui_backstore.py:434]: perfect!

**Review by pkalever — APPROVED** [review]

File: targetcli/ui_backstore.py [hunk 2]
```diff
@@ -423,17 +423,17 @@ def _create_file(self, filename, size, sparse=True):
             raise ExecutionError("Could not open %s" % filename)
         try:
             if sparse:
+                os.ftruncate(f.fileno(), size)
+            else:
+                self.shell.log.info("Writing %d bytes" % size)
                 try:
                     os.posix_fallocate(f.fileno(), 0, size)
                 except AttributeError:
                     # Prior to version 3.3, Python does not provide fallocate
```
Comment (pkalever) [inline targetcli/ui_backstore.py:432]: @maurizio-lombardi nit: just move the above comment line before call to posix_fallocate()
Rest all looks good.

**Pull request merged**